tests/src/test/scala/org/apache/openwhisk/core/containerpool/test/ContainerProxyTests.scala

```diff
@@ -1079,11 +1076,17 @@ class ContainerProxyTests
     def runCount = atomicRunCount.get()
     override def suspend()(implicit transid: TransactionId): Future[Unit] = {
       suspendCount += 1
-      super.suspend()
+      val s = super.suspend()
+      //verify that httpconn is closed
+      httpConnection should be(None)
+      s
     }
-    def resume()(implicit transid: TransactionId): Future[Unit] = {
+    override def resume()(implicit transid: TransactionId): Future[Unit] = {
       resumeCount += 1
-      Future.successful(())
+      val r = super.resume()
+      //verify that httpconn is recreated
+      httpConnection should be('defined)
```

Comment:
Please await the futures of either resume and suspend before checking that the client is there. Right now this is not an issue but once the implementation of the trait changes we're prone to heisenbugs.